libkbfs/block_retrieval_worker_test.go

```diff
@@ -251,3 +258,46 @@ func TestBlockRetrievalWorkerShutdown(t *testing.T) {
 	w.Shutdown()
 	require.True(t, shutdown)
 }
+
+func TestBlockRetrievalWorkerMultipleBlockTypes(t *testing.T) {
+	t.Log("Test that we can retrieve the same block into different block types.")
+	codec := kbfscodec.NewMsgpack()
+	q := newBlockRetrievalQueue(1, codec)
+	require.NotNil(t, q)
+	defer q.Shutdown()
+
+	bg := newFakeBlockGetter()
+	w1 := newBlockRetrievalWorker(bg, q)
+	require.NotNil(t, w1)
+	defer w1.Shutdown()
+
+	t.Log("Setup source blocks")
+	ptr1 := makeFakeBlockPointer(t)
+	block1 := makeFakeFileBlock(t)
+	ch1 := bg.setBlockToReturn(ptr1, block1)
+	testCommonBlock := &CommonBlock{}
+	bytes, err := codec.Encode(block1)
+	require.NoError(t, err)
+	err = codec.Decode(bytes, testCommonBlock)
+	require.NoError(t, err)
+
+	t.Log("Make a retrieval for the same block twice, but with a different target block type.")
+	testBlock1 := &FileBlock{}
+	testBlock2 := &CommonBlock{}
+	req1Ch := q.Request(context.Background(), 1, nil, ptr1, testBlock1)
+	req2Ch := q.Request(context.Background(), 1, nil, ptr1, testBlock2)
+	// Ensure the worker picks up the request
+	time.Sleep(50 * time.Millisecond)
```

Comment:
Oof, I'd rather not have sleeps in tests.  (I guess I missed the one in `TestBlockRetrievalWorkerWithQueue` on a previous review.)

Seems like you can get around this pretty easily by having `fakeBlockGetter.getBlock` notify an optional unbuffered channel that it's received a request.